src/background/stores/assessment-card-selection-store.ts

```diff
@@ -221,4 +252,90 @@ export class AssessmentCardSelectionStore extends PersistentStore<AssessmentCard
 
         await this.emitChanged();
     };
+
+    private onScanCompleted = async (payload: ScanCompletedPayload<any>) => {
+        const assessment = this.assessmentsProvider.forType(payload.testType);
+        if (!assessment) {
+            return;
+        }
+
+        const testKey = assessment.key;
+        const ruleId = payload.key;
+
+        if (!this.state[testKey]) {
+            this.state[testKey] = this.getDefaultTestCardSelectionData();
+        }
+
+        if (!this.state[testKey].rules) {
+            this.state[testKey].rules = {};
+        }
+
+        if (!this.state[testKey].rules![ruleId]) {
+            this.state[testKey].rules![ruleId] = {
+                cards: {},
+                isExpanded: false,
+            };
+        }
+
+        this.state[testKey].rules![ruleId].cards = this.createCardsFromSelectorMap(
+            ruleId,
+            payload.selectorMap,
+        );
+
+        await this.emitChanged();
+    };
+
+    private createCardsFromSelectorMap(
+        key: string,
+        selectorMap: DictionaryStringTo<HtmlElementAxeResults>,
+    ): CardSelectionData {
+        const cards: CardSelectionData = {};
+        forOwn(selectorMap, result => {
+            if (result.ruleResults && result.ruleResults[key] && result.ruleResults[key].id) {
+                cards[result.ruleResults[key].id!] = false;
+            }
+        });
+        return cards;
+    }
+
+    private onResetAllAssessmentsData = async (): Promise<void> => {
+        this.state = {};
```

Comment:
I'm okay with leaving this as is but I wonder if in the future we'll need to change this to 'go through each test key and just set it to default'. Regardless, I think `onScanCompleted` would be able to handle a completely empty object as well.